From 0e1608be9df30a3765d3c17ca01d7c5bfa542edd Mon Sep 17 00:00:00 2001
From: Vasiliy Ulyanov <vulyanov@suse.de>
Date: Thu, 22 Aug 2024 09:27:33 +0200
Subject: [PATCH] Consider the new DV reason ImagePullFailed

CDI v1.60.1 introduces a new reason ImagePullFailed for the DataVolume
Running condition. Take it into account to properly update the printable
status of a VM and to report the error.

Signed-off-by: Vasiliy Ulyanov <vulyanov@suse.de>
---
 pkg/storage/types/dv.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pkg/storage/types/dv.go b/pkg/storage/types/dv.go
index d90011815..a96984dc4 100644
--- a/pkg/storage/types/dv.go
+++ b/pkg/storage/types/dv.go
@@ -184,7 +184,7 @@ func HasDataVolumeErrors(namespace string, volumes []virtv1.Volume, dataVolumeSt
 		dvRunningCond := NewDataVolumeConditionManager().GetCondition(dv, cdiv1.DataVolumeRunning)
 		if dvRunningCond != nil &&
 			dvRunningCond.Status == v1.ConditionFalse &&
-			dvRunningCond.Reason == "Error" {
+			(dvRunningCond.Reason == "Error" || dvRunningCond.Reason == "ImagePullFailed") {
 			return fmt.Errorf("DataVolume %s importer has stopped running due to an error: %v",
 				volume.DataVolume.Name, dvRunningCond.Message)
 		}
-- 
2.46.0

